### DISCUSSION TITLE
how pass data branchId from loginPage to all Pages in My Project  by  session in asp.net core

### DISCUSSION THREAD

**Discussion by ra-bedo11** [serenity-is/Serenity#6109] [Help] [answered]
![login](https://user-images.githubusercontent.com/29665710/142774379-b050c10a-9406-4634-8125-3611d64cf9bc.jpg)
![customer](https://user-images.githubusercontent.com/29665710/142774175-f10e4140-3b65-43b7-b186-8bcc48bf701d.jpg)
**here in login page i use** 
this code after Check Success
HttpContext.Session.SetString("BranchId", branch.ToString());

because
I want at my project level to filter using by BranchId                   How is that ?

example

branch1
The first branch contains customers
----------------
branch2
The second branch contains other customers
------------------
etc ..............
------------
The first user who has validity on branch1 does not see the customers of branch2, and the user who has validity on branch 2 does not see the customers of the first branch etc.

**Comment by dfaruque** [comment]
See https://serenity.is/docs/tutorials/multi_tenancy/multi_tenancy

**Reply by ra-bedo11** [reply]
I do not want that

Is it possible to make a filter here with the session


![customer111](https://user-images.githubusercontent.com/29665710/142854391-9897251d-e9f9-4664-9449-5fa97baa18da.png)
?

**Reply by dfaruque** [reply]
you may apply filters in handlers.

**Comment by ra-bedo11** [comment, marked as answer]
![customer111](https://user-images.githubusercontent.com/29665710/142855954-e70cb685-c4fd-450a-b232-921a1bec7afa.png)

**Reply by dfaruque** [reply]
this is not the correct approch.

**Reply by dfaruque** [reply]
You have to do something like the following
```
        private class MyListHandler : ListRequestHandler<MyRow>
        {
            
            protected override void ApplyFilters(SqlQuery query)
            {
                base.ApplyFilters(query);

                var filterValue = 2; //you may use session here
                query.Where(fld.YourField == filterValue);
            }
        }
```

**Comment by ra-bedo11** [comment]
I already did it like this but the filter and search in Column doesn't work


‫في الاثنين، 22 نوفمبر 2021 في 3:01 م تمت كتابة ما يلي بواسطة ‪dfaruque‬‏ <‪
***@***.***‬‏>:‬

**Comment by ra-bedo11** [comment]
Thanks           its Work😐😐😐😐

‫في الاثنين، 22 نوفمبر 2021 في 4:58 م تمت كتابة ما يلي بواسطة علوم
الكمبيوتر ***@***.***‬‏>:‬

**Comment by ra-bedo11** [comment]
Thank u So Much
